src/Controller/AppController.php

```diff
@@ -153,14 +152,12 @@ public function beforeFilter(Event $event)
         } else if ($langInCookie) {
             $langInCookieAlias = $this->remapOldLangAlias($langInCookie);
             if ($langInCookieAlias != $langInCookie && !empty($langInURL)) {
-                $this->Cookie->write('CakeCookie.interfaceLanguage', $langInCookieAlias, false, "+1 month");
                 $langInCookie = $langInCookieAlias;
             }
             $lang = $langInCookie;
-        } else if (!empty($langInURL)) {
-            $lang = $langInURL;
-            $this->Cookie->write('CakeCookie.interfaceLanguage', $lang, false, "+1 month");
         }
+
+        $this->Cookie->write('CakeCookie.interfaceLanguage', $lang, false, "+1 month");
```

Comment:
Is the `CakeCookie` namespace really necessary?

CakePHP will send this cookie as `CakeCookie={"interfaceLanguage":"eng"}`, i.e. the browser stores it with the key `CakeCookie` and a JSON object as the value.

But on the Javascript side (when using the language chooser dialog) we [set the cookie as `CakeCookie[interfaceLanguage]=eng`](https://github.com/Tatoeba/tatoeba2/blob/98269d3af50a4b45e0c75456307bc13cbb5a9a7e/webroot/js/responsive/app.module.js#L101), i.e. the browser stores it as another cookie although CakePHP recognizes it for the same cookie (and even prefers this version).

But this is the only cookie for which we use the `CakeCookie` namespace so I wonder whether we could simply just use `interfaceLanguage` as the cookie key and thus avoid having two cookies stored for the same value.

Reply:
For historical reasons, but also because the namespace CakeCookie is configured as unencrypted. I think we should get rid of that JS code that sets the cookie as `CakeCookie[interfaceLanguage]=eng`, and have it set by the server using an action. It would avoid having to maintain two implementations to set that cookie.

Reply:
And it would allow to saving the UI language as a user setting. Anyway I think this is all well out of the scope of this pull request. :smile:

Reply:
> Anyway I think this is all well out of the scope of this pull request. smile

Agreed. I was just reminded of this duplication when I've reviewed the code.